pkg/controller/windowsmachineconfig/status.go

```diff
@@ -0,0 +1,103 @@
+	status             *wmcapi.WindowsMachineConfigStatus
+	reconciling        bool
+	degradationReasons map[string][]string
+}
+
+// NewStatusManager returns a new instance of StatusManager
+func NewStatusManager(client client.Client, name types.NamespacedName) *StatusManager {
+	return &StatusManager{
+		degradationReasons: make(map[string][]string),
+		client:             client,
+		name:               name,
+		status:             &wmcapi.WindowsMachineConfigStatus{},
+	}
+}
+
+// addDegradationReason adds the reason and message to the list of degradation causes
+func (s *StatusManager) addDegradationReason(reason, message string) {
+	messages, present := s.degradationReasons[reason]
+	if present {
+		s.degradationReasons[reason] = append(messages, message)
+	} else {
+		s.degradationReasons[reason] = []string{message}
+	}
+}
+
+// setReconciling updates the CR's status to indicate that the reconciler is active
+func (s *StatusManager) setReconciling() {
+	object := &wmcapi.WindowsMachineConfig{}
+	err := s.client.Get(context.TODO(), s.name, object)
+	if err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			return
+		}
+		log.Error(err, fmt.Sprintf("could not get %v", s.name))
+	}
+	wmcapi.SetWindowsMachineConfigCondition(&object.Status,
+		*wmcapi.NewWindowsMachineConfigCondition(wmcapi.Reconciling, corev1.ConditionTrue, "", ""))
+	if err = s.client.Status().Update(context.TODO(), object); err != nil {
+		log.Error(err, fmt.Sprintf("could not update status to %v", object.Status))
+	} else {
+		log.V(1).Info("updated status", "status", object.Status)
+	}
+}
+
+// updateStatus syncs the current status with the cluster and removes the reconciling status condition
+func (s *StatusManager) updateStatus() {
+	object := &wmcapi.WindowsMachineConfig{}
+	err := s.client.Get(context.TODO(), s.name, object)
+	if err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			return
+		}
+		log.Error(err, fmt.Sprintf("could not get %v", s.name))
+	}
+	object.Status.JoinedVMCount = s.status.JoinedVMCount
+	degraded := s.getDegradedCondition()
+	wmcapi.SetWindowsMachineConfigCondition(&object.Status, *degraded)
+	wmcapi.SetWindowsMachineConfigCondition(&object.Status,
+		*wmcapi.NewWindowsMachineConfigCondition(wmcapi.Reconciling, corev1.ConditionFalse, "", ""))
+	if err = s.client.Status().Update(context.TODO(), object); err != nil {
+		log.Error(err, fmt.Sprintf("could not update status to %v", object.Status))
+	} else {
+		log.Info("updated status", "status", object.Status)
+	}
+}
+
+// getDegradedCondition returns a degraded condition defined by the degradationReasons present in the StatusManager
+func (s *StatusManager) getDegradedCondition() *wmcapi.WindowsMachineConfigCondition {
+	// If there are no degradation reasons the WMC is not degraded
+	if len(s.degradationReasons) == 0 {
+		return wmcapi.NewWindowsMachineConfigCondition(wmcapi.Degraded, corev1.ConditionFalse, "", "")
+	}
+
+	// Add all reasons separated by a comma
```

Comment:
Thats different than the schema we discussed in planning:
This is the doc @ravisantoshgudimetla  put his suggested schema in:
https://docs.google.com/document/d/1OaeoYfnXHrLrTFeuU9EwWCKPtVCZdyAVFDIYPZAeDG0

Reply:
I did not realize there could be multiple reasons and degraded messages. The doc seems to imply singular. Now I am confused as to how there could be many reasons and messages for a particular condition.

Reply:
How can the operator be in degraded state because of multiple reasons? Shouldn't it enter degraded state the moment it sees first failure? Or you checking the current state and if it is degraded you're combining the current state with previous state? Or is that because we're continuing without capturing the error?

Reply:
This is happening because we attempt to do as much as possible instead of stopping at the first error. Because of this we need to keep track of all the errors that occur. Other operators that I've looked at stopped at the first error and didnt have this problem.